src/client/sandbox/fetch.js

```diff
@@ -121,7 +123,12 @@ export default class FetchSandbox extends SandboxBase {
 
             sandbox.emit(sandbox.FETCH_REQUEST_SENT_EVENT, fetchPromise);
 
-            return fetchPromise;
+            return fetchPromise
+                .then(response => {
+                    sandbox.cookieSandbox.syncServerCookie();
```

Comment:
I guess we should perform cookie synchronization before sending fetch request (before call `nativeMethods.fetch.apply(this, args);`)